src/error.rs

```diff
@@ -97,29 +97,166 @@ impl From<OddLengthStringError> for ToBytesError {
     fn from(e: OddLengthStringError) -> Self { Self::OddLengthString(e) }
 }
 
-/// Invalid hex character.
+/// UTF-8 character is not an ASCII hex character.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct InvalidCharError {
-    pub(crate) invalid: u8,
+    pub(crate) invalid: char,
     pub(crate) pos: usize,
 }
 
 impl InvalidCharError {
-    /// Returns the invalid character byte.
-    pub fn invalid_char(&self) -> u8 { self.invalid }
-    /// Returns the position of the invalid character byte.
+    /// Returns the invalid character.
+    pub fn invalid_char(&self) -> char { self.invalid }
+
+    /// Returns the position in the input string of the invalid character.
     pub fn pos(&self) -> usize { self.pos }
 }
 
 impl fmt::Display for InvalidCharError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "invalid hex char {} at pos {}", self.invalid_char(), self.pos())
+        write!(
+            f,
+            "char {} at pos {} is not a valid ASCII hex character",
+            self.invalid_char(),
+            self.pos()
+        )
     }
 }
 
 #[cfg(feature = "std")]
 impl std::error::Error for InvalidCharError {}
 
+/// Invalid ASCII encoding of a hex digit.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct InvalidDigitError {
+    /// The high digit in the pair e.g., for 'ab' this is 'a'.
+    pub(crate) hi: u8,
+
+    /// The high digit in the pair e.g., for 'ab' this is 'b'.
+    pub(crate) lo: u8,
+
+    /// `true` if the invalid byte is the `hi` one, `false` if its the `lo` one.
+    pub(crate) is_hi: bool,
+
+    /// Position of the stat of the byte pair in the input string.
+    pub(crate) pos: usize,
+}
+
+impl InvalidDigitError {
+    /// Returns the invalid byte.
+    pub fn invalid_byte(&self) -> u8 {
```

Comment:
In ce28e9ba01cda818439b36108e072435715552fc:

Ok, here you have a error which holds two digits, which are supposed to comprise a byte but don't. You have a method called `invalid_byte` which returns *one of the two digits*. I think this name is confusing. It should be called `invalid_digit`.